updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -103,6 +107,19 @@ def job_group
       @dependency_group_engine.find_group(name: T.must(job.dependency_group_to_refresh))
     end
 
+    sig { params(group: Dependabot::DependencyGroup).void }
+    def mark_group_handled(group)
+      directories.each do |directory|
+        @current_directory = directory
+
+        # add the existing dependencies in the group so individual updates don't try to update them
+        add_handled_dependencies(dependencies_in_existing_pr_for_group(group).filter_map { |d| d["dependency-name"] })
+        # also add dependencies that might be in the group, as a rebase would add them;
+        # this avoids individual PR creation that immediately is superseded by a group PR supersede
+        add_handled_dependencies(group.dependencies.map(&:name))
+      end
+    end
+
     sig { params(dependency_names: T.any(String, T::Array[String])).void }
     def add_handled_dependencies(dependency_names)
```

Comment:
[unrelated] It seems this method doesn't really do anything with the passed in parameter?